step_four_1d_2d_3d/interpolate.h

```diff
@@ -209,53 +239,29 @@ void ti_cpu_upsample_linear(
     }
 
     // Add all constant offsets
-    scalar_t * src0 = src + offset0[0] + offset0[1];
-    scalar_t * src1 = src + offset0[0] + offset1[1];
-    scalar_t * src2 = src + offset1[0] + offset0[1];
-    scalar_t * src3 = src + offset1[0] + offset1[1];;
+    scalar_t * src_[4];
+    src_[0] = src + offset0[0] + offset0[1];
+    src_[1] = src + offset0[0] + offset1[1];
+    src_[2] = src + offset1[0] + offset0[1];
+    src_[3] = src + offset1[0] + offset1[1];
+
+    index_t * idx[2];
+    float * w[2 * out_ndims];
+    for (int i = 0; i < out_ndims; i ++) {
+      w[2 * i] = w0[i];
+      w[2 * i + 1] = w1[i];
+    }
```

Comment:
We should just create the `w` instead of using `w0` and `w1` in the top, but I've left this here to avoid doing more changes.

Same for the indexes as well.